gdctools/lib/common.py

```diff
@@ -133,6 +133,19 @@ def map_blank_to_na(csvfile):
     for row in csvfile:
         yield map(lambda f: f if f != '' else 'NA', row)
 
+def rearrange_columns(csvfile, col_order):
+    """
+csvfile : iterable of list
+col_order : list of int
+    E.g.: col_order = [0, 2, 3, 1] will cause column 1 of the input to be column 3 of the output.
+    """
+    len_expected = max(col_order) + 1
+    for row in csvfile:
+        if len(row) != len_expected:
```

Comment:
It's not making the test more permissive - if you've already made the change to have expected length be `max(col_order) + 1`, then the final piece of implementing the functionality to drop columns is changing the test.